pkg/controller/networkpolicy/clustergroup.go

```diff
@@ -206,34 +221,48 @@ func (n *NetworkPolicyController) syncInternalGroup(key string) error {
 		klog.Infof("Didn't find the ClusterGroup %s, skip processing of internal group", grp.Name)
 		return nil
 	}
-	selectorUpdated := n.processServiceReference(grp)
 	newMemberSet := controlplane.GroupMemberSet{}
-	if grp.Selector != nil {
-		// Find all Pods matching its selectors and update store.
-		pods, externalEntities := n.processSelector(*grp.Selector)
-		for _, pod := range pods {
-			if len(pod.Status.PodIPs) == 0 {
-				// No need to insert Pod IPAddress when it is unset.
-				continue
+	selectorUpdated := false
+	if len(grp.ChildGroups) > 0 {
+		for _, childGroupName := range grp.ChildGroups {
+			childGrpObj, found, _ := n.internalGroupStore.Get(childGroupName)
+			if !found {
+				klog.V(2).Infof("Internal group %s not found when processing child groups for %s", childGroupName, grp.Name)
 			}
-			newMemberSet.Insert(podToGroupMember(pod, true))
+			childGrp := childGrpObj.(*antreatypes.Group)
+			newMemberSet = newMemberSet.Union(childGrp.GroupMembers)
```

Comment:
@Dyanngg I'm thinking about how to support nested group with https://github.com/vmware-tanzu/antrea/pull/1937.  
For the solution that keeps the GroupMembers in internalGroup, I feel it's not very different from just calculating the members when syncing AppliedToGroup and AddressGroup, as a ClusterGroup can be converted to one AppliedToGroup and one AddressGroup at most, so it's just one vs two union calculation. And not having GroupMembers in InternalGroup can save some memory. I think you could need to make a little change in `getAppliedToWorkloads` and `populateAddressGroupMemberSet`.

For the first solution that adds nested group concept to the grouping interface, I think it's doable but it would make the implementation a little more complicated, and may make little difference if it's not GET too often or the number of child groups is small. Perhaps let's first calculate the union on client side first and then discuss whether it's worth to optimize?

Reply:
Sure let me work on another commit to rebase this on #1937.

Reply:
Done